src/components/Carousel.tsx

```diff
@@ -1,18 +1,169 @@
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+interface State {
+  infiniteAuto: boolean,
+  infinite: boolean;
+  itemWidth: number,
+  frameSize: number,
+  duration: number,
+  step: number;
+}
+
+interface Props {
+  images: string[];
+}
+
+class Carousel extends React.Component<Props, State> {
+  state = {
+    infiniteAuto: false,
+    infinite: false,
+    duration: 1000,
+    itemWidth: 130,
+    frameSize: 3,
+    step: 3,
+  };
+
+  frameSizeHandler = (e:React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ frameSize: Number(e.target.value) });
+  };
+
+  itemWidthHandler = (e:React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ itemWidth: Number(e.target.value) });
+  };
+
+  stepHandler = (e:React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ step: Number(e.target.value) });
+  };
+
+  durationHandler = (e:React.ChangeEvent<HTMLInputElement>) => {
+    this.setState((state) => ({
+      duration: state.infiniteAuto
+        ? Number(e.target.value) * 3
+        : Number(e.target.value),
+    }));
+  };
+
+  infiniteHandler = (e:React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      infinite: e.target.checked,
+      infiniteAuto: false,
+    });
+  };
+
+  autoInfiniteHandler = (e:React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      infiniteAuto: e.target.checked,
+      infinite: false,
+    });
+  };
+
+  render() {
+    const {
+      step,
+      frameSize,
+      itemWidth,
+      duration,
+      infinite,
+      infiniteAuto,
+    } = this.state;
+
+    return (
+      <div className="Carousel">
+        <Animation
+          step={step}
+          infinite={infinite}
+          infiniteAuto={infiniteAuto}
+          duration={duration}
+          itemWidth={itemWidth}
+          frameSize={frameSize}
+          images={this.props.images}
+        />
+
+        <div className="Carousel__check">
+          <label htmlFor="infinite" className="Carousel__check--lable">
+            Infinite steps
+          </label>
+          <input
+            className="Carousel__check--checkbox"
+            id="infinite"
+            name="infinite"
+            type="checkbox"
+            checked={infinite}
+            onChange={this.infiniteHandler}
+          />
+
+          <label htmlFor="autoInfinite" className="Carousel__check-lable">
+            Auto-infinite
+          </label>
+          <input
+            className="Carousel__check--checkbox"
+            id="autoInfinite"
+            name="autoInfinite"
+            type="checkbox"
+            checked={infiniteAuto}
+            onChange={this.autoInfiniteHandler}
+          />
+        </div>
+        <div className="Carousel__setting">
+          <div className="Carousel__setting--lable">
+            <label htmlFor="step" className="lable">Step:</label>
+            <label htmlFor="itemWidth" className="lable">Withdrow:</label>
```

Comment:
```suggestion
            <label htmlFor="itemWidth" className="lable">Withdraw:</label>
```
I guess it's better to name it 'Image size'